src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -567,7 +598,7 @@ protected virtual string GenerateDisplayName([NotNull] ModelMetadata metadata, s
             if (resolvedDisplayName == null)
             {
                 resolvedDisplayName =
-                    string.IsNullOrEmpty(htmlFieldName) ? string.Empty : htmlFieldName.Split('.').Last();
+                    string.IsNullOrEmpty(expression) ? string.Empty : expression.Split('.').Last();
```

Comment:
- checking `string.IsNullOrEmpty(expression)` so `Split()` will always return an `array` of minimum length `1`
- BTW there is a longstanding bug in this code (doesn't handle `model[42]` correctly) but that's not going to get fixed in this PR (if ever)